File: Special Day/GradingSystem/java/NnamdiGradingSystemApp.java
import java.util.ArrayList;

public class NnamdiGradingSystemApp{

    public static void main(String... nnamdi){

    java.util.Scanner inputCollector = new java.util.Scanner(System.in);
 
        ArrayList<String> studentAndScores = new ArrayList<>();

        System.out.print("Enter the amount of student enrolled: ");
        int amountOfStudents = inputCollector.nextInt();

        System.out.print("Enter the amount of Number of quizzes taken?.. ");
        int numberOfQuiz = inputCollector.nextInt();

        
        double average = 0;
        int [][] justScores = new int[amountOfStudents][numberOfQuiz];
        ArrayList<Integer> onlyScores = new ArrayList<>();
        int row = 0;
        for(;row < amountOfStudents; row++){
            String studentDisplay = String.format("Student %d%n", row+1);
            studentAndScores.add(studentDisplay);
            int column = 0;
            System.out.printf("----- Student %d -----%n",row+1);
            int total = 0;
            for(; column < numberOfQuiz; column++){
                
                System.out.print("Enter the score of the student: ");
                int quizScore = inputCollector.nextInt();
                boolean chekingStudentScoreIsValid = true;
                if(quizScore >= 0 && quizScore <= 100){
                    String convert = quizScore + "";
                    studentAndScores.add(convert);
                    justScores[row][column] = quizScore;
                    onlyScores.add(quizScore);
                    total+=quizScore;
                
                }
                else{
                    while(chekingStudentScoreIsValid){
                        System.out.println("Wrong input check that score is within the range of 0-100");
               
                        System.out.print("Please enter the valid score of the student: ");
                        quizScore = inputCollector.nextInt();
                        if(quizScore >= 0 && quizScore <= 100){
                            String convert = quizScore + "";
                            studentAndScores.add(convert);
                            justScores[row][column] = quizScore;
                            onlyScores.add(quizScore);
                            total+=quizScore;
                            chekingStudentScoreIsValid = false;
                        }
                    }
                }
            }   
        
            average = total/numberOfQuiz;
            String convert = average + "";
            studentAndScores.add(convert);

        }


//        System.out.println(studentAndScores);
//        System.out.println(java.util.Arrays.deepToString(justScores));
//        System.out.println(onlyScores);

        ArrayList<String> quizAverage = new ArrayList<String>();
        double largest = 0;
        double averages = 0;
        int questionWithHighestAverage = 0;
        int indexIncreament = numberOfQuiz;
        int index = 0;
        for (; index < numberOfQuiz; index++){
            if (amountOfStudents == 1){
                double question = onlyScores.get(index);
                averages = question/amountOfStudents;
                String convertAverageToString = averages + "";
                quizAverage.add(convertAverageToString);
                double averageTotal = averages;
                if(averageTotal > largest){
                    largest = averageTotal;
                    questionWithHighestAverage+=1;
                }
                    indexIncreament+=1;

            }
    
            else{
                double question = onlyScores.get(index) + onlyScores.get(indexIncreament);
                averages = question/amountOfStudents;
                String convertAverageToString = averages + "";
                quizAverage.add(convertAverageToString);
                double averageTotal = averages;
                if(averageTotal > largest){
                    largest = averageTotal;
                    questionWithHighestAverage+=1;
                }
                indexIncreament+=1;
            }
        }

        System.out.printf("""
========================= QUIZ GRADE REPORT =========================

STUDENT       """
                                );
        int displayQuizCount = 0;
        for(; displayQuizCount < numberOfQuiz; displayQuizCount++){
            System.out.printf("      QZ%d   ", displayQuizCount+1);
        }
        System.out.print("    AVG\n");
        System.out.print("-------------------------------------------------------------------------\n");
        

        int dummyForQuizAmount = numberOfQuiz;
        int incrementByQuiz = 1;
        int innerRow = 0;
        for(;innerRow < amountOfStudents; innerRow++){
            int displayScores = incrementByQuiz;
            System.out.printf("Student %d   ", innerRow+1);
            for(;displayScores < dummyForQuizAmount+2; displayScores++){
                System.out.printf("%s    ",studentAndScores.get(displayScores));
            }
            incrementByQuiz+=numberOfQuiz+3;
            dummyForQuizAmount+=numberOfQuiz+3;
            System.out.printf(" ");
        }

        System.out.printf("""

------------------------------------------------------------------------- %n"""
        );
        

        System.out.print("The average of ");
        int displayQuizzes = 0;
        for (;displayQuizzes < numberOfQuiz; displayQuizzes++){
            System.out.printf("Quiz%d, ",displayQuizzes +1);
        }

        System.out.println(quizAverage);
        System.out.printf("Best Quiz: Quiz %d (avg: %.2f)%n",questionWithHighestAverage, largest );
        
    }
}

File: Special Day/GradingSystem/python/nnamdi_grading_system_app.py

print("----------------------------------------------------------------- WELCOME TO LEARN HUB -----------------------------------------------------------------")

amount_of_student = int(input("Enter the amount of student enrolled: "))
amount_of_quizzes = int(input("Number of quizzes taken: "))

student_scores = []

only_scores = []

total = 0

for student_count in range(0, amount_of_student):
    name_of_student = f"Student {student_count+1}"
    student_scores.append(name_of_student)
    print(f"student {student_count+1}")
    for quiz_count in range (0, amount_of_quizzes):
        quiz_score = int(input("Enter the score of the student: "))
        cheking_student_score_isValid = True
        if(quiz_score >= 0 and quiz_score <= 100):
            student_scores.append(quiz_score)
            only_scores.append(quiz_score)
            total+=quiz_score
        else:
            while(cheking_student_score_isValid):
                print("Wrong input check that score is within the range of 0-100")
               
                quiz_score = int(input("Please enter the valid score of the student: "))
                if(quiz_score >= 0 and quiz_score <= 100):
                    student_scores.append(quiz_score)
                    only_scores.append(quiz_score)
                    total+=quiz_score
                    cheking_student_score_isValid = False                  
                
#        student_scores.append(quiz_score)
#        total+=quiz_score
    average = total/amount_of_quizzes
    student_scores.append(average)

print(student_scores)
print(only_scores)

quiz_average = []
largest = 0
averages = 0
question_with_highest_average = 0
index_increament = amount_of_quizzes
for index in range(0, amount_of_quizzes):
    question = only_scores[index] + only_scores[index_increament]
    averages = question/amount_of_student
    quiz_average.append(averages)
    total = averages
    if(total > largest):
        largest = total
        question_with_highest_average+=1
   
    index_increament+=1

print("""
========================= QUIZ GRADE REPORT =========================
  
""")
print("STUDENT", end="     ")

for display_count in range(amount_of_quizzes):
    print(f"QZ{display_count +1}", end="   ") 

print("   AVG")

print("-------------------------------------------------------------------------")

score_index = 1
stop_index = amount_of_quizzes
dummy_for_quiz_amount = amount_of_quizzes
increment_by_quiz = 0
for display_student in range(amount_of_student):
#    print(f"Student {display_student +1} ", end="  ")
    for display_scores in range(increment_by_quiz, dummy_for_quiz_amount+2):
#        print(f"{student_scores(score_index, stop_index)}")
        print(f"{student_scores[display_scores]}", end="    ")
    increment_by_quiz+=amount_of_quizzes+2
    dummy_for_quiz_amount+=amount_of_quizzes+2
    print()

    

print("-------------------------------------------------------------------------")
print("The average of", end=" ")
for display_quizzes in range(amount_of_quizzes):
    print(f"Quiz{display_quizzes+1}", end=",")

print(quiz_average)
print(f"Best Quiz: Quiz {question_with_highest_average} (avg:{largest})")
    
